src/modules/glaxnimate/producer_glaxnimate.cpp

```diff
@@ -141,12 +141,15 @@ static bool createQApplicationIfNeeded(mlt_service service)
 #endif
 #if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
         if (getenv("DISPLAY") == 0 && getenv("WAYLAND_DISPLAY") == 0) {
-            mlt_log_error(service,
+            const char *qt_qpa = getenv("QT_QPA_PLATFORM");
+            if (qt_qpa == 0 || strcmp(qt_qpa, "offscreen") != 0) {
```

Comment:
We prefer (!qt_qpa || strcmp(qt_qpa, “offscreen”))  rather than add additional, unnecessary comparisons.